### PR TITLE
Release monofile v1.4

### PR DESCRIPTION
**Pull request by nbitzz**
## monofile 1.4
- [ ] API v1 (classic API will be referred to as API v0) + docs
- [ ] Fix sync issues between account and files db (or just switch db tech entirely because json for db is :plague:)
- [ ] ability to generate API tokens to use for apps
- [ ] oauth (:URETHRA:)
- [X] longer tokens
- [X] Bearer authentication

### CODE REVIEW

**Comment by nbitzz** [comment]
@Jack5079 should we target a migration to sql+prisma in 2.0 or is that too much for this release?
the roadmap of sorts that i have envisioned in my mind is 2.0 > 2.1 (Markofile) > 3.0 (Prisma+Interchangeable&multiple file hosts) but idkkkkkk

this json database is a curse and i don't know what i was thinking using one but...idk

**Comment by Jack5079** [comment]
idk about markofile anymore but where would we host the sql

**Comment by Jack5079** [comment]
but yeah Too Much

**Comment by nbitzz** [comment]
> idk about markofile anymore but where would we host the sql

honestly i have 0 idea.probably offsite because monofile's flagships are on fucking GLITCH

**Comment by Jack5079** [comment]
> idk about markofile anymore but where would we host the sql

SQLite probably

**Comment by nbitzz** [comment]
> > idk about markofile anymore but where would we host the sql
> 
> SQLite probably

or that if we don't go with like postgres

**Comment by Jack5079** [comment]
> > idk about markofile anymore but where would we host the sql
> 
> honestly i have 0 idea.probably offsite because monofile's flagships are on fucking GLITCH

Drizzle ORM supports Cloudflare D1 but I have no idea if you can do that outside a Worker

**Comment by nbitzz** [comment]
i should probably remove some of these comments 😭
i posted about this on fedi but i'll post it here too
https://github.com/mollersuite/monofile/blob/0f9bcba740a7ef79f48eaf4f3a62c9805853662f/src/server/lib/DiscordAPI/DiscordRequests.ts#L3-L7

honestly most of monofile still needs heavy refactoring, i'll be working on it whenever i have time to do so

**Comment by nbitzz** [comment]
Up next is probably finishing api-v1 (finally)
I just wanna get it over with atp 😭

**Comment by nbitzz** [comment]
I think I'm going to revisit palettes and theming
Are we making the new default, like, Catppuccin Mocha? *Should* we even do that, or should we just make our own palette to match 1.3's original colors? I'm not sure /shrug

**Comment by nbitzz** [comment]
Catppuccin's style guide is at https://github.com/catppuccin/catppuccin/blob/main/docs/style-guide.md so I guess I'll just follow that for Making a Monofile

**Comment by nbitzz** [comment]
How would we do the red background for stuff like login errors...? @Jack5079

**Comment by nbitzz** [comment]
Could also use a thick border on the left as a tab of sorts... but I'm not sure if that fits the current UI

**Comment by nbitzz** [comment]
ABORT ABORT ABORT FUCKING ABORT GOD THE CLIENT IS SO ASS

**Comment by nbitzz** [comment]
TO DO (numbered in order) (not final)
1. #69
2. #54
3. #16
4. #68
5. #18
6. #12 (Release 🎉)

TL;DR realistically monofile 2.0 is not done until Late Q2. Which is really funny for fucking Discord File Uploader but... yeah

Why am i doing all this for what is essentially a joke project 😭 3.0 (making monofile a serious project) needs to come ASAP after 2.0
